libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFNetwork.m

```diff
@@ -41,6 +45,41 @@ @implementation SFNetwork
 static NSURLSessionConfiguration *kSFSessionConfig;
 __weak static id<SFNetworkSessionManaging> kSFNetworkManager;
 
++ (instancetype)defaultEphemeralNetwork {
+    NSURLSessionConfiguration *sessionConfiguration = [NSURLSessionConfiguration ephemeralSessionConfiguration];
+    return [[self alloc] initWithSessionIdentifier:kSFNetworkEphemeralSessionIdentifier sessionConfiguration:sessionConfiguration];
+}
+
++ (instancetype)defaultBackgroundNetwork {
+    NSURLSessionConfiguration *sessionConfiguration = [NSURLSessionConfiguration backgroundSessionConfigurationWithIdentifier:kSFNetworkBackgroundSessionIdentifier];
+    return [[self alloc] initWithSessionIdentifier:kSFNetworkBackgroundSessionIdentifier sessionConfiguration:sessionConfiguration];
+}
+
++ (instancetype)networkWithSessionIdentifier:(nonnull NSString *)identifier sessionConfiguration:(nullable NSURLSessionConfiguration *)sessionConfiguration {
+    return [[self alloc] initWithSessionIdentifier:identifier sessionConfiguration:sessionConfiguration];
```

Comment:
I don't think we should `alloc` a new instance of this class every time it's called. Instead, you want to look up the `static` dictionary to figure out if an instance exists with the session identifier passed in. If so, return it. If not, then we `alloc` and put that new instance in the `static` dictionary. Here is a similar example from `SFSDKSalesforceAnalyticsManager` (it would look similar to this):
```
+ (instancetype) sharedInstanceWithUser:(SFUserAccount *) userAccount {
    @synchronized ([SFSDKSalesforceAnalyticsManager class]) {
        if (!userAccount) {
            userAccount = [SFUserAccountManager sharedInstance].currentUser;
        }
        if (!userAccount) {
            return nil;
        }
        NSString *key = SFKeyForUserAndScope(userAccount, SFUserAccountScopeCommunity);
        if (!key) {
            return nil;
        }
        id analyticsMgr = analyticsManagerList[key];
        if (!analyticsMgr) {
            if (userAccount.loginState != SFUserAccountLoginStateLoggedIn) {
                [SFSDKCoreLogger w:[self class] format:@"%@ A user account must be in the  SFUserAccountLoginStateLoggedIn state in order to create a SFSDKSalesforceAnalyticsManager instance for a user.", NSStringFromSelector(_cmd)];
                return nil;
            }
            analyticsMgr = [[self alloc] initWithUser:userAccount];
            analyticsManagerList[key] = analyticsMgr;
        }
        return analyticsMgr;
    }
}
```